src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -274,13 +277,27 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
   handleFieldChange = (name: string, value: unknown) => {
     if (['timeWindow', 'aggregation'].includes(name)) {
       this.setState({[name]: value});
+      if (name === 'aggregation') {
+        if (value === 0) {
+          this.props.rule.aggregate = 'count()';
+        } else if (value === 1) {
+          this.props.rule.aggregate = 'count_unique(tags[sentry:user])';
+        }
+      }
```

Comment:
This is the "hack" that keeps the new backend aggregate synced up to any changes a user makes.